libnmstate/nm/team.py

```diff
@@ -17,14 +17,81 @@
 # along with this program. If not, see <https://www.gnu.org/licenses/>.
 #
 
-from libnmstate.error import NmstateNotImplementedError
+import json
+
+from libnmstate.nm import connection as nm_connection
+from libnmstate.nm import nmclient
+from libnmstate.schema import Interface
 from libnmstate.schema import Team
 
 
-def create_setting(iface_state):
-    setting = None
+def create_setting(iface_state, base_con_profile):
+    team_setting = None
     team_config = iface_state.get(Team.CONFIG_SUBTREE)
-    if team_config:
-        raise NmstateNotImplementedError
+    if base_con_profile:
+        if team_config:
+            team_setting = base_con_profile.get_setting_duplicate(
+                nmclient.NM.SETTING_TEAM_SETTING_NAME
+            )
+
+            if not team_setting:
+                team_setting = nmclient.NM.SettingTeam.new()
+
+            team_config = _convert_team_config_to_setting_format(
+                team_config, iface_state.get(Interface.NAME)
+            )
+
+            team_setting.props.config = json.dumps(team_config)
+
+    return team_setting
+
+
+def _convert_team_config_to_setting_format(team_config, ifname):
+    team_config["device"] = ifname
+
+    if team_config.get(Team.PORT_SUBTREE):
```

Comment:
You can fetch it once into a variable so you will not need to fetch it twice.

Reply:
Done, thanks!